indico/modules/rb_new/operations/bookings.py

```diff
@@ -209,3 +214,44 @@ def get_booking_occurrences(booking):
     date_range = sorted(set(cand.start_dt.date() for cand in booking.occurrences))
     occurrences = group_by_occurrence_date(booking.occurrences)
     return date_range, occurrences
+
+
+def check_room_available(room, start_dt, end_dt):
+    occurrences = get_existing_room_occurrences(room, start_dt, end_dt, allow_overlapping=True)
+    prebookings = [occ for occ in occurrences if not occ.reservation.is_accepted]
+    bookings = [occ for occ in occurrences if occ.reservation.is_accepted]
+    unbookable_hours = get_rooms_unbookable_hours([room])
+    hours_overlap = [hours for hours in unbookable_hours
+                     if overlaps((start_dt.time(), end_dt.time()), (hours.start_time, hours.end_time))]
+    nonbookable_periods = get_rooms_nonbookable_periods([room], start_dt, end_dt)
+    blockings = get_rooms_blockings([room], start_dt, end_dt).get(room.id, [])
+    blocked_for_user = [blocking for blocking in blockings
+                        if not blocking.blocking.can_be_overridden(session.user, room)]
+    user_bookings = [booking for booking in bookings if booking.reservation.booked_for_id == session.user.id]
+    user_prebookings = [prebooking for prebooking in prebookings
+                        if prebooking.reservation.booked_for_id == session.user.id]
+
+    return {
+        'can_book': room.can_be_booked(session.user),
+        'can_prebook': room.can_be_prebooked(session.user),
+        'conflict_booking': bool(bookings),
+        'conflict_prebooking': bool(prebookings),
+        'unbookable': bool(hours_overlap or nonbookable_periods or blocked_for_user),
+        'user_booking': bool(user_bookings),
+        'user_prebooking': bool(user_prebookings),
+    }
+
+
+def create_booking_for_event(room_id, event):
+    try:
+        room = Room.get_one(room_id)
+        default_timezone = timezone(config.DEFAULT_TIMEZONE)
+        start_dt = event.start_dt.astimezone(default_timezone).replace(tzinfo=None)
+        end_dt = event.end_dt.astimezone(default_timezone).replace(tzinfo=None)
+        booking_reason = "Event '%s'" % (event.title)
```

Comment:
i18n?

Reply:
No, we don't translate log messages

Reply:
Isn't it the reason of the booking?

Reply:
ah indeed. still, it's visible to everyone so IMO the same applies

Reply:
or should we just use the event title without "Event '...'" around it?

Reply:
yeah, maybe it's the easiest solution

Reply:
Just a use case, if I have my Indico instance in Polish I'd find strange ending up with a partially English booking reason.

Reply:
As a non-polish room owner I'd find it strange to have a polish justification why someone's using my room :)

We don't take the user's language into account anywhere else where content will become visible to others than the current user (e.g. emails)

Reply:
I've used `Event '...'` to indicate it's booked for the event itself. I figured we will have `Contribution '...'` and `Session block '...'` for the same event as well.